cicada/commands/exec_schedule.py

```diff
@@ -153,109 +153,112 @@ def main(schedule_id, dbname=None):
+        try:
+            # pylint: disable=consider-using-with
+            child_process = subprocess.Popen(
+                full_command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+            )
+
+            # Check if child process has terminated
+            while returncode is None:
+                time.sleep(1)
+                returncode = child_process.poll()
+
+                # If still running, check if child_process should be aborted
+                if returncode is None:
+                    # protect against db unavailable
                     try:
                         db_conn = postgres.db_cicada(dbname)
                         db_cur = db_conn.cursor()
-                        db_connection_made = True
-                    except Exception:
+                        if get_abort_running(db_cur, schedule_id):
+
+                            # Terminate main process
+                            returncode = -15
+                            error_detail = "Cicada abort_running"
+                            unset_abort_running(db_cur, schedule_id)
+                        db_cur.close()
+                        db_conn.close()
+                    # pylint: disable=unused-variable
+                    except Exception as error:
                         send_slack_error(
                             schedule_id,
                             schedule_log_id,
                             returncode,
-                            "Cicada database not reachable",
+                            "Cicada database not reachable to check abort_running",
                         )
                         time.sleep(1)
 
-                unset_is_running(db_cur, schedule_id)
-                finalize_schedule_log(db_cur, schedule_log_id, returncode, error_detail)
+        # Capture error
+        except OSError as error:
+            returncode = error.errno
+            error_detail = error.strerror
+        except subprocess.CalledProcessError as error:
+            returncode = error.returncode
+            error_detail = "CalledProcessError"
+        except KeyboardInterrupt:
+            returncode = 1
+            error_detail = "KeyboardInterrupt"
+        except SystemExit:
+            returncode = 1
+            error_detail = "SystemExit"
+        except Exception:
+            returncode = 999
+            error_detail = "Crazy Unknown Error"
+        finally:
+
+            # Terminate any zombie processes
+            for zombie in psutil.Process(cicada_pid).children(recursive=True):
+                zombie.send_signal(signal.SIGTERM)
+
+            # Repeatedly attempt to finalize schedule, even if db is unavailable
+            db_connection_made = False
+            while not db_connection_made:
+                try:
+                    db_conn = postgres.db_cicada(dbname)
+                    db_cur = db_conn.cursor()
+                    db_connection_made = True
```

Comment:
This connection is not being closed anywhere

Reply:
Now we know why the script originally had a close at the end ... reverting to that